src/sentry/integrations/request_buffer.py

```diff
@@ -0,0 +1,109 @@
+import logging
+from datetime import datetime, timedelta
+
+from django.conf import settings
+
+from sentry.utils import json, redis
+
+BUFFER_SIZE = 30  # 30 days
+KEY_EXPIRY = 60 * 60 * 24 * 30  # 30 days
+
+IS_BROKEN_RANGE = 7  # 7 days
+
+logger = logging.getLogger(__name__)
+
+
+class IntegrationRequestBuffer:
+    """
+    Create a data structure to store daily successful and failed request counts for each installed integration in Redis
+    This should store the aggregate counts of each type for last 30 days for each integration
+    """
+
+    def __init__(self, integration_id):
+        self.integration_id = integration_id
+
+        cluster_id = settings.SENTRY_INTEGRATION_ERROR_LOG_REDIS_CLUSTER
+        self.client = redis.redis_clusters.get(cluster_id)
+
+    def _get_redis_key(self):
+
+        return f"sentry-integration-error:{self.integration_id}"
+
+    def _convert_obj_to_dict(self, redis_object):
+        """
+        Convert the request string stored in Redis to a python dict
+        """
+
+        return json.loads(redis_object)
+
+    def _get_all_from_buffer(self, buffer_key):
+        """
+        Get the list at the buffer key.
+        """
+
+        return self.client.lrange(buffer_key, 0, BUFFER_SIZE - 1)
+
+    def _get(self):
+        """
+        Returns the list of daily aggregate error counts.
+        """
+        return [
+            self._convert_obj_to_dict(obj)
+            for obj in self._get_all_from_buffer(self._get_redis_key())
+        ]
+
+    def is_integration_broken(self):
+        """
+        Integration is broken if we have 7 consecutive days with a non-zero error count
+        """
+        data = [
+            datetime.strptime(item.get("date"), "%Y-%m-%d").date()
+            for item in self._get()
```

Comment:
Should `_get()` only return the last `IS_BROKEN_RANGE` days? Since we do not look at the rest of the 30 days?